libheif/heif_colorconversion.cc

```diff
@@ -2948,8 +2949,9 @@ bool ColorConversionPipeline::construct_pipeline(ColorState input_state,
 
 void ColorConversionPipeline::debug_dump_pipeline() const
 {
-  for (const auto& op : m_operations) {
-    std::cerr << "> " << typeid(*op).name() << "\n";
+  for (const auto& op_ptr : m_operations) {
+    auto& op = *op_ptr;
+    std::cerr << "> " << typeid(op).name() << "\n";
```

Comment:
This is the second of the two warning fixes in this PR.

All the other changes in this PR are intended to maintain a consistent style in the code that operates on 'ops' or m_operations. Please feel free to implement an alternative fix to suit your preference. Thank you!